### ISSUE TITLE
Difference not rendered correctly in preview when camera is inside object

### ISSUE THREAD

**Issue by Masterhigure** [openscad/openscad#4748]
As it says in the title. Working in version 2021.01, when previewing a difference, if the camera is inside one of the objects that are subtracted, it is no longer subtracted.

**To Reproduce**
The code
```
difference() {
    sphere(30);
    cube([1000, 1000, 1000]);
}
```
should be plenty. When I render preview (F5), and move the camera to the first octant (inside the huge cube), then the sphere appears whole. When I render properly (F6), the issue disappears.

**Screenshots**
![Camera outside the first quadrant, sphere missing a wedge](https://github.com/openscad/openscad/assets/41048478/1c3159e5-bd85-4740-9d16-7a194e6ce9a5)
![Camera inside the first quadrant, sphere whole](https://github.com/openscad/openscad/assets/41048478/b55eb596-47ff-4913-964b-03e1f79420c1)

**Environment and Version info (please complete the following information):**
 - OS: Windows 11
 - System: AMD Ryzen 3 5425U with Radeon Graphics 2.70 GHz
 - OpenSCAD Version 2021.01, installed from the x86-64.exe installer downloaded from openscad.org

**Library information**
```
OpenSCAD Version: 2021.01
System information: Windows(TM) 10.0 SP 0.0 NTW 1 MSDN 724833 8 CPUs 15.31 GB RAM
User Agent: OpenSCAD/2021.01 (Windows(TM) 10.0 SP 0.0 NTW 1 MSDN 724833)
Compiler: GCC "7.4.0" 64bit
MinGW build: MingW64
Debug build: No
Boost version: 1_60
Eigen version: 3.3.7
CGAL version, kernels: 4.11, Cartesian<Gmpq>, Extended_cartesian<Gmpq>, Epeck
OpenCSG version: OpenCSG 1.4.2
Qt version: 5.13.0
QScintilla version: 2.11.2
InputDrivers: 
GLib version: 2.50.2
lodepng version: 20180910
libzip version: 1.5.2
fontconfig version: 2.13.1
freetype version: 2.10.1
harfbuzz version: 2.4.0
cairo version: 1.16.0
lib3mf version: 1.8.0
```

**Comment by nophead** [comment]
This has always been the case unless you switch to Orthogonal mode or used render().  In Perceptive mode anything in front of the front clipping plane gets removed. So the subtracted object gets clipped and no longer subtracts. I don't think it is fixable as it is a feature of the OpenCSG library used for preview.

**Comment by jordanbrown0** [comment]
Yeah, this is pretty much just the way it is, absent a total revamp of the preview pipeline.

It's documented [in the FAQ](https://en.wikibooks.org/wiki/OpenSCAD_User_Manual/FAQ#Why_does_difference_(or_intersection)_sometimes_not_work_in_preview?)
